binderhub/tests/test_build.py

```diff
@@ -241,6 +241,48 @@ def test_git_credentials_passed_to_podspec_upon_submit():
     assert env["GIT_CREDENTIAL_ENV"] == git_credentials
 
 
+def test_extra_environment_variables_passed_to_podspec_upon_submit():
+    extra_environments = {
+        "CONTAINER_HOST": "unix:///var/run/docker.sock",
+        "REGISTRY_AUTH_FILE": "/root/.docker/config.json",
+    }
+
+    mock_k8s_api = _list_dind_pods_mock()
+
+    class EnvBuild(Build):
```

Comment:
The `Build` class is deprecated and is only kept for backwards compatibility. Are you able to inherit from `BuildExecutor` instead, and configure it using traitlets?

Reply:
In this case, `KubernetesBuildExecutor` shall be used as the tested changed is implemented there but that said, the test is working using it and traitlets.